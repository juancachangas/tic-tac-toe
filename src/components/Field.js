/*
 * Replace the `alert` with the actual dispatch later on
 *
 * You already have utilit classes available to mark the current field selected by a player:
 *  .is-player-0
 *  .is-player-1
 *
 * Check out the src/index.html for pre-defined basic css classes.
 */
import React from 'react';

const Field = ({
  id,
  checkedBy = undefined,
}) => {
  const isPlayer = checkedBy && checkedBy < 2 ? `is-player-${checkedBy}` : '';
  return (
    <div
      className={`Field ${isPlayer}`}
      role="presentation"

    >
      {id}
    </div>
  );
};

export default Field;
