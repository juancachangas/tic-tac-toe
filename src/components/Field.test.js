import React from 'react';
import { shallow } from 'enzyme';

import Field from './Field';

describe('Field component is', () => {
  it('rendering default', () => {
    const wrapper = shallow(
      <Field id="1" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('rendering player 0 move', () => {
    const wrapper = shallow(
      <Field id="1" checkedBy="0" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('rendering player 1 move', () => {
    const wrapper = shallow(
      <Field id="1" checkedBy="1" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('not rendering player move made by other players', () => {
    const wrapper = shallow(
      <Field id="1" checkedBy="2" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
