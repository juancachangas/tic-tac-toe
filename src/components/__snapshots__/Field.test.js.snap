// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component is not rendering player move made by other players 1`] = `
<div
  className="Field "
  role="presentation"
>
  1
</div>
`;

exports[`Field component is rendering default 1`] = `
<div
  className="Field "
  role="presentation"
>
  1
</div>
`;

exports[`Field component is rendering player 0 move 1`] = `
<div
  className="Field is-player-0"
  role="presentation"
>
  1
</div>
`;

exports[`Field component is rendering player 1 move 1`] = `
<div
  className="Field is-player-1"
  role="presentation"
>
  1
</div>
`;
